cardano-api/src/Cardano/Api/Block.hs

```diff
@@ -105,6 +125,21 @@ instance Show (Block era) where
         . showsPrec 11 block
         )
 
+getBlockTxs :: forall era . Block era -> [Tx era]
+getBlockTxs (ByronBlock Consensus.ByronBlock { Consensus.byronBlockRaw }) =
+    case byronBlockRaw of
+      Byron.ABOBBoundary{} -> [] -- no txs in EBBs
+      Byron.ABOBBlock Byron.ABlock {
+          Byron.blockBody =
+            Byron.ABody {
+              Byron.bodyTxPayload = Byron.ATxPayload txs
+            }
+        } -> map ByronTx txs
+getBlockTxs (ShelleyBlock shelleyEra Consensus.ShelleyBlock{Consensus.shelleyBlockRaw})
+  = case shelleyEra of
+      ShelleyBasedEraShelley -> case shelleyBlockRaw of Shelley.Block _header (Shelley.TxSeq txs) -> [ShelleyTx shelleyEra x | x <- toList txs]
+      ShelleyBasedEraAllegra -> case shelleyBlockRaw of Shelley.Block _header (Shelley.TxSeq txs) -> [ShelleyTx shelleyEra x | x <- toList txs]
+      ShelleyBasedEraMary    -> case shelleyBlockRaw of Shelley.Block _header (Shelley.TxSeq txs) -> [ShelleyTx shelleyEra x | x <- toList txs]
```

Comment:
I wonder if we can do this in a parametrised way for all ledger eras, and then call it for each era (ie keeping the GADT pattern match, but dispatch to the same parametrised impl). For most other cases in the lib we've been able to use generic impls for the Shelley-based eras.